lib/rpmsg/rpmsg.c

```diff
@@ -305,7 +305,7 @@ void rpmsg_register_endpoint(struct rpmsg_device *rdev,
 			     rpmsg_ept_cb cb,
 			     rpmsg_ns_unbind_cb ns_unbind_cb, void *priv)
 {
-	strncpy(ept->name, name ? name : "", sizeof(ept->name));
+	strlcpy(ept->name, name ? name : "", sizeof(ept->name));
```

Comment:
I wonder if we should create our own strncpy.
we could create a `libutil/strncpy.c` that contains a `safe_strncpy` function.

```c
char *safe_strncpy(char *dest, char *src, size_t size) {
	size_t len;

	if (size == 0) {
		dest[0] = '\0';
	} else {	
		len = strnlen(src, size - 1);
		memcpy(dest, src, len);
		dest[len] = '\0';
	}
	return dest;
}

```
@edmooring, @tnmysh : any opinon/suggestion?

Reply:
I believe that makes sense. In kernel as well, strlcpy was deprecated, and strscpy was introduced. https://www.kernel.org/doc/html/latest/core-api/kernel-api.html?highlight=declare_bitmap#c.strscpy

Reply:
I am okay, to introduce such function.